src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -583,38 +591,37 @@ std::shared_ptr<ExpressionResult<Result>> ExpressionEvaluator::_evaluate_case_ex
     const CaseExpression& case_expression) {
   const auto when = evaluate_expression_to_result<ExpressionEvaluator::Bool>(*case_expression.when());
 
-  std::shared_ptr<ExpressionResult<Result>> result;
+  std::vector<Result> values;
+  std::vector<bool> nulls;
 
   _resolve_to_expression_results(
       *case_expression.then(), *case_expression.otherwise(), [&](const auto& then_result, const auto& else_result) {
         using ThenResultType = typename std::decay_t<decltype(then_result)>::Type;
         using ElseResultType = typename std::decay_t<decltype(else_result)>::Type;
 
-        const auto result_size = _result_size(when->size(), then_result.size(), else_result.size());
-        std::vector<Result> values(result_size);
-        std::vector<bool> nulls(result_size);
-
         // clang-format off
-      if constexpr (CaseEvaluator::supports_v<Result, ThenResultType, ElseResultType>) {
-        for (auto chunk_offset = ChunkOffset{0};
-             chunk_offset < result_size; ++chunk_offset) {
-          if (when->value(chunk_offset) && !when->is_null(chunk_offset)) {
-            values[chunk_offset] = to_value<Result>(then_result.value(chunk_offset));
-            nulls[chunk_offset] = then_result.is_null(chunk_offset);
-          } else {
-            values[chunk_offset] = to_value<Result>(else_result.value(chunk_offset));
-            nulls[chunk_offset] = else_result.is_null(chunk_offset);
+        if constexpr (CaseEvaluator::supports_v<Result, ThenResultType, ElseResultType>) {
+          const auto result_size = _result_size(when->size(), then_result.size(), else_result.size());
+          values.resize(result_size);
+          nulls.resize(result_size);
+
+          for (auto chunk_offset = ChunkOffset{0};
```

Comment:
unintended linebreak?

Reply:
There is no need to turn of the formatter there any more, anyway.

Reply:
There is no need to turn of the formatter there any more, anyway.